documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -0,0 +1,517 @@
+  query API,
+- `org.hibernate.validator:hibernate-validator` and
+  `org.glassfish:jakarta.el` if you're using Hibernate Validator,
+- `org.hibernate:query-validator` to enable compile-time checking of your
+  HQL queries, and/or
+- the Hibernate bytecode enhancer if you want to use field-level lazy
+  fetching.
+
+TIP: Field-level lazy fetching is an advanced feature that most programs
+don't need. Stick to the basics for now.
+
+There's an example {build}[Gradle build] included in the example program.
+
+=== Configuration
+
+:xml: https://github.com/hibernate/hibernate-reactive/blob/master/example/src/main/resources/META-INF/persistence.xml
+
+Hibernate Reactive is configured via the standard JPA `persistence.xml`
+document which must be placed, as usual, in the `/META-INF` directory.
+
+The only configuration specific to Hibernate Reactive is the persistence
+`<provider>` element, which must be explicit:
+
+[source,xml]
+----
+<provider>org.hibernate.reactive.provider.ReactivePersistenceProvider</provider>
+----
+
+Otherwise, configuration is almost completely transparent. Configure
+Hibernate exactly as you usually would, noting that most configuration
+properties related to JDBC or JTA aren't relevant in the context of
+Hibernate Reactive.
+
+Configuration properties of particular interest include:
+
+- `javax.persistence.jdbc.url`, the JDBC URL of your database,
+- `javax.persistence.jdbc.user` and `javax.persistence.jdbc.password`,
+the database credentials,
+- `javax.persistence.schema-generation.database.action`, for enabling
+  automatic schema export.
+
+An example {xml}[`persistence.xml`] file is included in the example
+program.
+
+=== Logging the generated SQL
+
+:log4j: https://github.com/hibernate/hibernate-reactive/blob/master/example/src/main/resources/log4j.properties
+
+To see the generated SQL as it's sent to the database, either:
+
+- set the property `hibernate.show_sql` to `true`, or
+- enable debug-level logging for the category `org.hibernate.SQL`
+  in for your preferred SLF4J logging implementation.
+
+For example, if you're using Log4J via `slf4j-log4j12` (as above in
+<<_optional_dependencies>>), add this line to your `log4j.properties`
+file:
+
+[source,properties]
+----
+log4j.logger.org.hibernate.SQL=DEBUG
+----
+
+An example {log4j}[`log4j.properties`] file is included in the example
+program.
+
+=== Tuning the Vert.x pool
+
+The Vert.x database client has built-in connection pooling and prepared
+statement caching.
+
+- Use `hibernate.connection.pool_size` to set the size of the reactive
+  connection pool.
+
+When it comes time for performance tuning, you can further customize the
+pool and cache via the following properties:
+
+- `hibernate.vertx.pool.max_wait_queue_size`
+- `hibernate.vertx.prepared_statement_cache.max_size`
+- `hibernate.vertx.prepared_statement_cache.sql_limit`
+
+TIP: But for now, just leave these three settings alone.
```

Comment:
I don't understand this TIP. Why mention them and then say that they shouldn't be set without any explanation?

Reply:
Is it because it is an introduction? If that's the case can you mention this property in a separate section and add a link here to it?

Reply:
Just because a brand new user doesn't need to actually *make use* of a feature right now, doesn't mean that they don't need to know that *it exists*, so that they will know what to reach for when the time comes.

Also, we don't currently have a reference doc, and I have absolutely zero intention of writing and maintaining one at this stage, so this introduction needs to serve a double purpose for now.

Reply:
Can we just remove the TIP then? I think this sentence explain it clearly:
```
When it comes time for performance tuning, you can further customize the
pool and cache via the following properties:
```